### PR TITLE
add volume type and metadata options in create_volume

### PR DESCRIPTION
**Pull request by zhitongLBN**
according to http://developer.openstack.org/api-ref-compute-v2.1.html   --> /v2.1/​{tenant_id}​/os-volumes
added `volume_type` and `metadata` in `compute/create_volume` option

### CODE REVIEW

File: lib/fog/openstack/requests/compute/create_volume.rb [hunk 1]
```diff
@@ -11,21 +13,28 @@ def create_volume(name, description, size, options={})
...
+
+          vanilla_options.select { |o| options[o] }.each do |key|
             data['volume'][key] = options[key]
           end
+
           request(
-            :body     => Fog::JSON.encode(data),
-            :expects  => [200, 202],
-            :method   => 'POST',
-            :path     => "os-volumes"
```
Comment (Ladas) [inline lib/fog/openstack/requests/compute/create_volume.rb]: please keep the hash rocket notation =>

**Comment by zhitongLBN** [comment]
done :)

**Comment by zhitongLBN** [comment]
please configure rubocop syntax preference :)

**Comment by Ladas** [comment]
@zhitongLBN A second, I just activated it, sorry for the spam

For some reason it takes the default from houndci, instead of our .rubocop.yml

**Comment by Ladas** [comment]
@zhitongLBN the hound is not listening :-) please ignore the rubocop change for now

**Comment by Ladas** [comment]
@zhitongLBN alright, seems like it started to work, but you might need to change something and push -f, so it restarts the hound job

**Comment by Ladas** [comment]
@zhitongLBN sorry for the complications :-)

**Comment by zhitongLBN** [comment]
@Ladas I did a rebase and push -f

**Comment by Ladas** [comment]
@zhitongLBN those seems like real rubocop issues according to the rules we have set, but the double quote is not required now so you can delete that commit and you should get less conflicts

Btw. if you will think some of the rules are more obtrusive than helpful, start an issue for the change of the rules and we can discuss that :-)

**Comment by zhitongLBN** [comment]
@Ladas adapted to new rubocop style

**Comment by Ladas** [comment]
@zhitongLBN looks good, thank you :-)

**Pull request merged**